### ISSUE TITLE
google.auth.getProjectId returns incorrent project id

### ISSUE THREAD

**Issue by evil-shrike** [googleapis/google-auth-library-nodejs#1211]
I use setting global options for auth as described in the docs - https://github.com/googleapis/google-api-nodejs-client#setting-global-or-service-level-auth

I execute my server with a cli argument `keyFile` that contains a path to a service account's key file. I pass that key file to google.options to set up auth globally. And it's used by all googleapi's clients for all APIs automatically. But when I call `google.auth.getProjectId()` it returns not a project from the keyFile (that I expect) but a project set via `gcloud config set project`.

```ts
import { google } from 'googleapis';

  let keyFile = argv.keyFile;
  const auth = new google.auth.GoogleAuth({
    keyFile: keyFile,
    scopes: OAUTH_SCOPES
  });
  google.options({
    auth: auth
  });
  let projectId = await google.auth.getProjectId();
  // wil be arbitrary project set in the system via gcloud config set
  projectid = await auth.getProjectId();
  // will be the expected one - from the keyFile
```

#### Environment details

  - OS: MacOS 11.4
  - Node.js version: v14.5.0
  - npm version:
  - `googleapis` version: 81

**Comment by tmatsuo** [comment]
@evil-shrike 

FYI I transferred this issue here. I think this is the right repo for this issue.

The project id is determined by:
https://github.com/googleapis/google-auth-library-nodejs/blob/dd2d07a01daf9d7e766189b889fb36a9f6fb8e73/src/auth/googleauth.ts#L190

Do you have any of the environment variables like these?
https://github.com/googleapis/google-auth-library-nodejs/blob/dd2d07a01daf9d7e766189b889fb36a9f6fb8e73/src/auth/googleauth.ts#L638

`getFileProjectId` is the second in the list. It should be prioritized.

Also are you unblocked by using `auth.getProjectId()`?

**Comment by evil-shrike** [comment]
Thanks, @tmatsuo 
`auth.getProjectId()` works as expected but the idea of using the global auth setup is to avoid dealing with auth in all other places. So I don't have direct access to that auth object where I need to get project id. I found a workaround though:
```ts
projectId = await (<any>google._options.auth).getProjectId();
```
but it's a bit hacky.

**Comment by tmatsuo** [comment]
@evil-shrike I think this will be fixed with https://github.com/googleapis/nodejs-common/pull/663

**Comment by tmatsuo** [comment]
@evil-shrike The fix is in the 3.7.1 release. Feel free to reopen if you still see the problem.
